parser/shared/src/main/scala/jawn/ByteArrayParser.scala

```diff
@@ -40,11 +40,29 @@ final class ByteArrayParser[J](src: Array[Byte]) extends SyncParser[J] with Byte
     context: FContext[J],
     stack: List[FContext[J]]
   ): Unit = {}
-  final protected[this] def byte(i: Int): Byte = src(i)
-  final protected[this] def at(i: Int): Char = src(i).toChar
 
-  final protected[this] def at(i: Int, k: Int): CharSequence =
+  final protected[this] def byte(i: Int): Byte = {
+    if (Platform.isJs) {
+      if (i < 0 || i >= src.length) throw new ArrayIndexOutOfBoundsException
+    }
+    src(i)
+  }
```

Comment:
I decided that platform-specific guards would be the best way to implement this.

Because `isJs` is a compile-time constant, the Scala compiler will actually elide these checks on JVM+Native before it even reaches the bytecode. So there is zero-cost to JVM and Native due to this change.

The other option is that we split these sources across platform but I think that is worse for readability.

Reply:
> Because `isJs` is a compile-time constant, the Scala compiler will actually elide these checks on JVM+Native before it even reaches the bytecode.

til. Is this eliding part of the Scala 2 inliner / Scala 3 `inline` keyword or just part of core scalac?

Its ifs the former maybe it makes sense to explicitly inline using `@inline`/`inline` keyword respectively so its more clearly?

In any case maybe it makes sense to add this as a brief comment to all of the `Platform` files for the different platforms?